src/sentry/integrations/opsgenie/integration.py

```diff
@@ -120,7 +121,46 @@ def dispatch(self, request: Request, pipeline) -> HttpResponse:  # type:ignore
 
 
 class OpsgenieIntegration(IntegrationInstallation):
-    pass
+    def get_client(self) -> Any:
+        return OpsgenieClient(integration=self.model)
+
+    def get_organization_config(self) -> Sequence[Any]:
+        fields = [
+            {
+                "name": "team_table",
+                "type": "table",
+                "label": "Opsgenie teams with the Sentry integration enabled",
+                "help": "If teams need to be updated, deleted, or added manually please do so here. Alert rules will need to be individually updated for any additions or deletions of teams.",
+                "addButtonText": "",
+                "columnLabels": {"team": "Team", "integration_key": "Integration Key"},
+                "columnKeys": ["team", "integration_key"],
+                "confirmDeleteMessage": "Any alert rules associated with this team will stop working. The rules will still exist but will show a `removed` team.",
+            }
+        ]
+
+        return fields
+
+    def update_organization_config(self, data: MutableMapping[str, Any]) -> None:
+        client = self.get_client()
+        # get the team ID/test the API key for a newly added row
+        # potential problem: the user could be rate limited if they hand-type every API key...
+        # another problem: the default error message isn't very helpful, but I don't know how to change it
+        teams = data["team_table"]
```

Comment:
@leeandher could I get your opinion about the update flow? The current flow is as such: the user enters the team name and their API key. If the API key is correct, the team ID is added to the config table and the table saves; otherwise, there is an error message and the team is not added to the config data. My concern about this is that if the user decides to type each key manually for whatever reason, they could get rate limited due to the instantaneous nature of the config update. 

Also, the default error message ("unable to save") doesn't provide much information. Do you think that we should make it more informative, or is it fine as is?

Reply:
That's a good callout, and TBH it's an issue with a lot of onBlur UX but I dont think we need to do anything about it here. I'd say because the same issue exists for Pagerduty, its alright to retain the behaviour, especially because I dont believe many people will be manually entering keys. As for the error though, I'd look into a way to surface a custom error indicating why we can't save it.

'Unable to save' sounds to me like an internal error, if we can tell the user something like 'Authorization Error from Opsgenie' or something more informative, it may help them realize the user error.

Reply:
@leeandher added a more useful error message.

![Uploading Screenshot 2023-07-17 at 2.13.38 PM.png…]()

Reply:
@leeandher added a more useful error message.

<img width="1512" alt="Screenshot 2023-07-17 at 2 15 16 PM" src="https://github.com/getsentry/sentry/assets/83109586/fed74608-c6b4-48e3-be01-88edf8dfb816">